### PR TITLE
Add API Test For Selected Shipping Method functionality

### PR DESCRIPTION
**Pull request by 06romix**
### Fixed Issues (if relevant)

1. magento/graphql-ce#478: Selected Shipping Method' functionality

### Manual testing scenarios (*)

1. N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=491) <br/>All committers have signed the CLA.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SelectedShippingMethodTest.php [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        $selectedShippingMethod = $this->getSelectedShippingMethod($response);
+
+        self::assertEquals('flatrate', $selectedShippingMethod['carrier_code']);
+        self::assertEquals('flatrate', $selectedShippingMethod['method_code']);
+        self::assertEquals('Flat Rate - Fixed', $selectedShippingMethod['label']);
+        self::assertEquals(0, $selectedShippingMethod['amount']);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_address.php
```
Comment (TomashKhamlai) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SelectedShippingMethodTest.php]: I guess there should be annotation like
`* @expectedException \Magento\Framework\Exception\NoSuchEntityException`

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SelectedShippingMethodTest.php [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        $selectedShippingMethod = $this->getSelectedShippingMethod($response);
+
+        self::assertEquals('flatrate', $selectedShippingMethod['carrier_code']);
+        self::assertEquals('flatrate', $selectedShippingMethod['method_code']);
+        self::assertEquals('Flat Rate - Fixed', $selectedShippingMethod['label']);
+        self::assertEquals(0, $selectedShippingMethod['amount']);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_address.php
```
Comment (TomashKhamlai) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SelectedShippingMethodTest.php]: The same thing here

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SelectedShippingMethodTest.php [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+{
+    /**
+     * @var \Magento\TestFramework\ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @var \Magento\Framework\Api\SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer/SelectedShippingMethodTest.php]: Please, clear all of not used dependencies

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SelectedShippingMethodTest.php [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    /**
+     * Retrieve quote by given reserved order ID
+     *
+     * @param string $reservedOrderId
+     * @return \Magento\Quote\Model\Quote
+     * @throws \InvalidArgumentException
+     */
+    protected function getCart($reservedOrderId)
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SelectedShippingMethodTest.php]: Looks like we don't need this method

**Review by naydav — CHANGES_REQUESTED** [review]

**Pull request merged**